code/WorkInProgress/buildmode.dm

```diff
@@ -6,50 +6,38 @@
 			log_admin("[key_name(usr)] has left build mode.")
 			M.client.buildmode = 0
 			M.client.show_popup_menus = 1
-			/*
-			var/list/bmholders = list()
-			for(var/obj/effect/bmode/buildholder/H)
-				if(H.cl == M.client)
-					//del(H)
-					bmholders += H
-			for(var/obj/effect/bmode/bm)
-				for(var/obj/effect/bmode/buildholder/HH in bmholders)
-					if(bm.master == HH) del(bm)
-			for(var/obj/effect/bmode/buildholder/HH in bmholders)
-				del(HH)
-				*/
 			var/obj/effect/bmode/buildholder/holder = null
-			for(var/obj/effect/bmode/buildholder/H)
+			for(var/obj/effect/bmode/buildholder/H in buildmodeholders)
 				if(H.cl == M.client)
 					holder = H
 					break
 			if(holder) holder.buildmode.copycat = null
 			if(M.client.buildmode_objs && M.client.buildmode_objs.len)
 				for(var/BM in M.client.buildmode_objs)
-					del(BM)
+					returnToPool(BM)
 		else
 			log_admin("[key_name(usr)] has entered build mode.")
 			M.client.buildmode = 1
 			M.client.show_popup_menus = 0
 
-			var/obj/effect/bmode/buildholder/H = new/obj/effect/bmode/buildholder()
-			var/obj/effect/bmode/builddir/A = new/obj/effect/bmode/builddir(H)
+			var/obj/effect/bmode/buildholder/H = getFromPool(/obj/effect/bmode/buildholder)
+			var/obj/effect/bmode/builddir/A = getFromPool(/obj/effect/bmode/builddir,H)

```

Comment:
frankly they could have just been initialized directly into the vars of the buildmode holder